deployment/helm/resource-management-policies/balloons/templates/configmap_enable_nri.yaml

```diff
@@ -0,0 +1,31 @@
+{{- if .Values.nri.enable }}
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: patch-containerd-config
+  namespace: {{ .Release.Namespace }}
+  labels:
+    {{- include "balloons-plugin.labels" . | nindent 4 }}
+data:
+  patch_nri.sh: |
+   #!/bin/bash
+
+   config_file="/home/config.toml" 
+   cp /etc/containerd/config.toml "$config_file"
+
+   # Check if the disable field is set to false under [plugins."io.containerd.nri.v1.nri"]
+   status=$(awk -F ' = ' '/^\[plugins\."io\.containerd\.nri\.v1\.nri"\]/{flag=1} flag && /disable/{print $2; exit}' "$config_file")
```

Comment:
Wouldn't it be simpler to just grep if `[plugins."io.containerd.nri.v1.nri"]` exists, do sed unconditioinally and use e.g. diff to see if anything was changed? WDYT?

Plus, in either of these cases there's still the corner case that the`[plugins."io.containerd.nri.v1.nri"]` section is present but the the `disable` setting hasn't been specified explicitly. Do we need to cover this(?) 🧐

Reply:
> Wouldn't it be simpler to just grep if `[plugins."io.containerd.nri.v1.nri"]` exists, do sed unconditioinally and use e.g. diff to see if anything was changed? WDYT?
> 
It can be simpler, but it also means extra/unnecessary restart of the containerd.

> Plus, in either of these cases there's still the corner case that the`[plugins."io.containerd.nri.v1.nri"]` section is present but the the `disable` setting hasn't been specified explicitly. Do we need to cover this(?) monocle_face

Is that a valid config? I mean, does containerd accept a config file without disable field set to either true/false?

Reply:
> So, diff would be used I guess to check if restart is needed or not right? Would not it be better instead of having diff just to check if we actually need to touch the file or not? Actually for me these two options (1. what I have, 2. what you suggest) don't really make a much difference. One is doing diff before touching the file the other is after touching the file.

Sure, if the diff is empty don't update the file. I was thinking about (in pseudo'ish bash cod)

```
sed ... /etc/containerd/config.toml > config.tmp
if ! diff -q /etc/containerd/config.toml config.tmp; then 
  mv config.tmp /etc/containerd/config.toml
  systemctl restart containerd
fi
```

> > Plus, in either of these cases there's still the corner case that the`[plugins."io.containerd.nri.v1.nri"]` section is present but the the `disable` setting hasn't been specified explicitly. Do we need to cover this(?) monocle_face
> 
> Is that a valid config? I mean, does containerd accept a config file without disable field set to either true/false?

Yeah, it is. The config may be an empty file or just one line if at the extrema.